sources/engine/Stride.Engine/Engine/Splines/Processors/SplineProcessor.cs

```diff
@@ -0,0 +1,152 @@
+//// Copyright (c) Stride contributors (https://Stride.com)
+//// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Stride.Core.Mathematics;
+using Stride.Engine.Splines.Components;
+using Stride.Engine.Splines.HierarchyTransformOperations;
+using Stride.Engine.Splines.Models;
+using Stride.Graphics;
+using Stride.Rendering;
+
+namespace Stride.Engine.Splines.Processors
+{
+    /// <summary>
+    /// The processor for <see cref="SplineComponent"/>.
+    /// </summary>
+    public class SplineProcessor : EntityProcessor<SplineComponent, SplineProcessor.SplineTransformationInfo>
+    {
+        private HashSet<SplineComponent> splineComponentsToUpdate = new();
+        private SplineBuilder splineBuilder;
+        private SplineRenderer SplineRenderer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SplineProcessor"/> class.
+        /// </summary>
+        public SplineProcessor()
+            : base(typeof(TransformComponent))
+        {
+            splineBuilder = new SplineBuilder();
+            SplineRenderer = new SplineRenderer();
+        }
+
+        protected override SplineTransformationInfo GenerateComponentData(Entity entity, SplineComponent component)
+        {
+            var transformationInfo = new SplineTransformationInfo(this, component) { TransformOperation = new SplineViewHierarchyTransformOperation(component) };
+
+            return transformationInfo;
+        }
+
+        protected override bool IsAssociatedDataValid(Entity entity, SplineComponent component, SplineTransformationInfo associatedData)
+        {
+            return component == associatedData.TransformOperation.SplineComponent;
+        }
+
+        protected override void OnEntityComponentAdding(Entity entity, SplineComponent component, SplineTransformationInfo data)
+        {
+            component.Spline.OnSplineDirty += data.OnSplineDirtyAction;
+            splineComponentsToUpdate.Add(component);
+            entity.Transform.PostOperations.Add(data.TransformOperation);
+        }
+
+        protected override void OnEntityComponentRemoved(Entity entity, SplineComponent component, SplineTransformationInfo data)
+        {
+            //When a spline component is removed, we still need to destroy the Spline renderer
+            DestroySplineRenderer(null, entity);
+            component.Spline.OnSplineDirty -= data.OnSplineDirtyAction;
+            entity.Transform.PostOperations.Remove(data.TransformOperation);
+        }
```

Comment:
Probably better to add `splineComponentsToUpdate.Remove(component)` in case it gets added and removed without a frame running in between (it can happen if components are manually removed after a load, etc.)